src/Controllers/MoviesController.cs

```diff
@@ -38,36 +49,53 @@ public IActionResult Index(string movieGenre, string searchString, string sortOr
             {
                 movies = movies.Where(m => m.Genre == movieGenre);
             }
-            if (!String.IsNullOrEmpty(sortOrder))
-            {
-                switch (sortOrder)
-                {
-                    case "Title":
-                        movies = movies.OrderBy(m => m.Title);
-                        break;
-                    case "Release Date":
-                        movies = movies.OrderBy(m => m.ReleaseDate);
-                        break;
-                    case "Genre":
-                        movies = movies.OrderBy(m => m.Genre);
-                        break;
-                    default:
-                        movies = movies.OrderBy(m => m.Rating);
-                        break;
-                }
+            switch (sortOrder)
+            {
+                case "title_desc":
+                    movies = movies.OrderByDescending(m => m.Title);
+                    break;
+                case "date":
+                    movies = movies.OrderBy(m => m.ReleaseDate);
+                    break;
+                case "date_desc":
+                    movies = movies.OrderByDescending(m => m.ReleaseDate);
+                    break;
+                case "genre":
+                    movies = movies.OrderBy(m => m.Genre);
+                    break;
+                case "genre_desc":
+                    movies = movies.OrderByDescending(m => m.Genre);
+                    break;
+                default:
+                    movies = movies.OrderBy(m => m.Title);
+                    break;
             }
             var movieGenreVM = new MovieGenreViewModel();
-            movieGenreVM.movies = movies.ToList();
-            movieGenreVM.genres = new List<SelectListItem>();
+            movieGenreVM.CurrentFilter = searchString;
+            movieGenreVM.CurrentSortOrder = sortOrder;
+            movieGenreVM.CurrentMovieGenre = movieGenre;
+            var movieVMs = movies.Select(
+            m => new MovieViewModel
```

Comment:
we usually use only one line for this: `var movieVMs = movies.Select(m => new MovieViewModel`